replica/database.hh

```diff
@@ -595,7 +595,9 @@ private:
     storage_group* storage_group_for_id(size_t i) const;
 
     std::unique_ptr<storage_group_manager> make_storage_group_manager();
+public:
     compaction_group* get_compaction_group(size_t id) const noexcept;
+private:
```

Comment:
ideally we won't expose compaction group to the outside world, which is an implementation detail confined to replica::table. 

we can instead introduce: table_state& try_get_table_state_with_static_sharding. if not in static sharding mode, then we throw an exception.